src/pages/settings/Wallet/ReportCardLostPage.tsx

```diff
@@ -135,8 +105,27 @@ function ReportCardLostPage({
             setShouldShowAddressError(true);
             return;
         }
+        setIsValidateCodeActionModalVisible(true);
+    };
+
+    const sendValidateCode = () => {
+        const primaryLogin = account?.primaryLogin ?? '';
+
+        if (loginList?.[primaryLogin]?.validateCodeSent) {
+            return;
+        }
 
-        CardActions.requestReplacementExpensifyCard(physicalCard.cardID, reason?.key as ReplacementReason);
+        requestValidateCodeAction();
+    };
+
+    const handleValidateCodeEntered = (validateCode: string) => {
+        CardActions.requestReplacementExpensifyCard(physicalCard.cardID, reason?.key as ReplacementReason, validateCode).then((newCardID) => {
+            if (!newCardID) {
+                return;
+            }
+            setIsValidateCodeActionModalVisible(false);
+            Navigation.navigate(ROUTES.SETTINGS_WALLET_DOMAINCARD.getRoute(newCardID));
+        });
```

Comment:
It doesn't look like we did this navigation before, is there a reason we are adding it now?